main/src/library/List.flix

```diff
@@ -0,0 +1,748 @@
+        case (x1, x2) :: rs => let (r1, r2) = unzip(rs);
+                                   (x1 :: r1, x2 :: r2)
+    }
+
+    //
+    // ## Two List Operations
+    //
+
+    ///
+    /// Returns a list where the element at index `i` is `f(a, b)` where
+    /// `a` is the element at index `i` in `xs` and `b` is the element at index `i` in `ys`.
+    /// If either `xs` or `ys` becomes depleted, then no further elements are added to the resulting list.
+    /// Functionally equivalent to zipWith.
+    ///
+    def map2[a,b,c](f: (a, b) -> c, xs: List[a], ys: List[b]): List[c] = zipWith(f, xs, ys)
+
+    ///
+    /// Concatenates the results of applying `f` pairwise to the elements of `xs` and `ys`.
+    ///
+    def flatMap2[a,b,c](f: (a, b) -> List[c], xs: List[a], ys: List[b]): List[c] = match (xs, ys) with {
+        case (x :: rs, y :: qs) => append(f(x, y), flatMap2(f, rs, qs))
+        case _ => Nil
+    }
+
+    ///
+    /// Accumulates the result of applying `f` pairwise to the elements of `xs` and `ys`
+    /// starting with the initial value `c` and going from left to right.
+    /// Alias for foldLeft2.
+    ///
+    def fold2[a,b,c](f: (c, a, b) -> c, c: c, xs: List[a], ys: List[b]): c = foldLeft2(f, c, xs, ys)
+
+    ///
+    /// Accumulates the result of applying `f` pairwise to the elements of `xs` and `ys`
+    /// starting with the initial value `c` and going from left to right.
+    ///
+    def foldLeft2[a,b,c](f: (c, a, b) -> c, c: c, xs: List[a], ys: List[b]): c = match (xs, ys) with {
+        case (x :: rs, y :: qs) => foldLeft2(f, f(c, x, y), rs, qs)
+        case _ => c
+    }
+
+    ///
+    /// Accumulates the result of applying `f` pairwise to the elements of `xs` and `ys`
+    /// starting with the initial value `c` and going from right to left.
+    ///
+    def foldRight2[a,b,c](f: (a, b, c) -> c, c: c, xs: List[a], ys: List[b]): c = match (xs, ys) with {
+        case (x :: rs, y :: qs) => foldLeft2(f, f(x, y, c), rs, qs)
+        case _ => c
+    }
+
+    //
+    // ## Combined Operations
+    //
+
+    ///
+    /// Returns the result of applying `f` to every element in `xs` and concatenating the results.
+    /// Alias for flatMap.
+    ///
+    def concatMap[a,b](f: a -> List[b], xs: List[a]): List[b] = flatMap(f, xs)
+
+    ///
+    /// Collects the results of applying the partial function `f` to every element in `xs`.
+    ///
+    def filterMap[a,b](f: a -> Option[b], xs: List[a]): List[b] = match xs with {
+        case Nil => Nil
+        case x :: rs => match f(x) with {
+                            case None => filterMap(f, rs)
+                            case Some(v) => v :: filterMap(f, rs)
+        }
+    }
+
+    ///
+    /// Returns the first non-None result of applying the partial function `f` to each element of `xs`.
+    /// Returns `None` if every element of `xs` is `None`.
+    ///
+    def findMap[a,b](f: a -> Option[b], xs: List[a]): Option[b] = match xs with {
```

Comment:
`findMap` should combine find and map. So you give it a function `a -> Option[b]` when this function returns `Some(...)` we will have found the thing we are interested in. So yeah, its look correct to me.